### PR TITLE
Fixes #1630 - BP: delete batches on Batch Status tab

### LINKED ISSUE #1630: BP: delete batches on Batch Status tab
On each row on the Batch Status table, there will be a trash can icon (this will be added as part of #1594). 

When the user clicks the trash can icon, prompt the user with an alert box that says "Are you sure you want to delete this batch?" If the user confirms, delete the batch by doing the following:

Make a DELETE request to the BatchProcessor (https://streamstats.usgs.gov/batchprocessor) /batch/{deleteCode} endpoint (example: /batch/24474b12-71c0-46f3-a149-b735ea833e26)

The deleteCode is a special sequence of characters that is returned by the GET request to the BatchProcessor (https://streamstats.usgs.gov/batchprocessor) /batch/ endpoint that is used in #1594.

### PR DESCRIPTION
**Pull request by amedenblik**
Closes #1630

Please wait to review until this merge request has been merged and deployed https://code.usgs.gov/StreamStats/web-services-and-apis/ss-batchprocessor/-/merge_requests/42

### CODE REVIEW

File: src/Controllers/BatchProcessorController.js [hunk 1]
```diff
@@ -373,6 +374,24 @@ var StreamStats;
                 }).finally(function () {
                 });
             };
+            BatchProcessorController.prototype.deleteBatch = function (batchID, deleteCode, batchStatusEmail) {
+                var _this = this;
+                var text = "Are you sure you want to delete Batch ID " + batchID + "?";
+                if (confirm(text) == true) {
+                    var url = configuration.baseurls['BatchProcessorServices'] + configuration.queryparams['SSBatchProcessorDeleteBatch'].format(deleteCode);
```
Comment (alaws-USGS) [inline src/Controllers/BatchProcessorController.js]: The 'SSBatchProcessorDeleteBatch' end point needs to be added to the 'configuration.queryparams' in 'appConfig.js' before I can test

**Review by alaws-USGS — CHANGES_REQUESTED** [review]

File: src/Controllers/BatchProcessorController.js (hunk 1 above)
Comment (amedenblik) [reply src/Controllers/BatchProcessorController.js]: Woops, sorry! Added it

File: src/Controllers/BatchProcessorController.ts [hunk 2]
```diff
@@ -640,6 +643,28 @@ module StreamStats.Controllers {
                 });
         }
 
+        public deleteBatch(batchID: number, deleteCode: string, batchStatusEmail: string) {
+            let text = "Are you sure you want to delete Batch ID " + batchID + "?"
+            if (confirm(text) == true) {
+                var url = configuration.baseurls['BatchProcessorServices'] + configuration.queryparams['SSBatchProcessorDeleteBatch'].format(deleteCode);
```
Comment (alaws-USGS) [inline src/Controllers/BatchProcessorController.ts:654]: Just to keep with convention, can you separate the service call from the calling function and return an ng.Ipromise from the service call.

**Review by alaws-USGS — CHANGES_REQUESTED** [review]

**Review by alaws-USGS — APPROVED** [review]
This looks great, thank you for making the changes!

**Pull request merged**